apiserver/common/crossmodel/crossmodel.go

```diff
@@ -35,18 +35,24 @@ func PublishRelationChange(backend Backend, relationTag names.Tag, change params
 		return errors.Trace(err)
 	}
 
-	// Update the relation status if necessary.
-	relStatus := status.Status(change.Status)
-	if relStatus != "" {
-		currentStatus, err := rel.Status()
-		if err != nil {
+	// Update the relation suspended status.
+	currentStatus := rel.Suspended()
+	if change.Suspended != nil && currentStatus != *change.Suspended {
+		if err := rel.SetSuspended(*change.Suspended); err != nil {
 			return errors.Trace(err)
 		}
-		if currentStatus.Status != relStatus || currentStatus.Message != change.StatusMessage {
+		if *change.Suspended {
```

Comment:
maybe dedup the below, so it looks like:

```go
st := status.Suspending
action := "suspending"
if !*change.Suspended {
    st = status.Joining
    action = "resuming"
}
<same code here, using st & action>
```

Reply:
done